### PR TITLE
fixes #10865 - do not fail to load the server if qpid listener fails

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by jlsherrill** [comment]
@dustints   @iNecas do you guys know any reason why this behaviour would have changed?  I just started seeing it recently

File: lib/katello/engine.rb [hunk 1]
```diff
@@ -34,8 +34,12 @@ class Engine < ::Rails::Engine
     end
 
     initializer "katello.initialize_cp_listener", after: "foreman_tasks.initialize_dynflow" do
-      unless ForemanTasks.dynflow.config.remote? || File.basename($PROGRAM_NAME) == 'rake' || Rails.env.test?
-        ForemanTasks.async_task(::Actions::Candlepin::ListenOnCandlepinEvents)
+      begin
+        unless ForemanTasks.dynflow.config.remote? || File.basename($PROGRAM_NAME) == 'rake' || Rails.env.test?
```
Comment (stbenjam) [inline lib/katello/engine.rb]: Foreman has a helper: Foreman.in_rake?

**Comment by iNecas** [comment]
I don't think just logging the exception and ignoring it is a way to go, as the action is not currently running, just the task state is in inconsisntent state.

I will open a PR on foreman tasks to ensure the tasks have proper state, so that we don't hit this issue

**Comment by jlsherrill** [comment]
Yeah, i agree, closing

**Comment by iNecas** [comment]
@jlsherrill could you review/test https://github.com/theforeman/foreman-tasks/pull/113 ?

**Pull request closed without merging**